### PR TITLE
Linter take 2

### PR DESCRIPTION
**Pull request by petorsong**
[Notion task](https://www.notion.so/uwblueprintexecs/make-eslint-actually-usable-f273baf9f5d14581ade84128bacef3c0)

Lints all changes in `paramedics-react` up to commit 76ef31c (CCP dashboard screen 1)

## Changes
- reinstalled eslint and prettier with new settings
  - added new rules after going through all code
- made further changes to fix specific rules (inline comments are provided)

## Testing
-

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -1,5 +1,6 @@
 {
   "printWidth": 80,
-  "singleQuote": false,
+  "singleQuote": true,
```
Comment (petorsong) [inline .prettierrc:3]: changed strings to single quotes - most of the diff is probably just this

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      "error",
+      "ignorePackages",
+      {
+        "js": "never",
+        "jsx": "never",
+        "ts": "never",
+        "tsx": "never"
+      }
+    ],
+    "@typescript-eslint/indent": "off",
```
Comment (petorsong) [inline .eslintrc:28]: eslint rules that were turned off

File: src/components/CCPDashboard/PatientOverview.tsx [hunk 3]
```diff
@@ -13,20 +12,22 @@ import {
...
-import { ScanPatientButton } from "./ScanPatientButton";
-import LoadingState from "../common/LoadingState";
+} from '../../graphql/queries/patients';
+import { TriageTag } from './TriageTag';
+import { PatientInfoTable } from './PatientInfoTable';
+import { ScanPatientButton } from './ScanPatientButton';
+import LoadingState from '../common/LoadingState';
 
 interface PatientOverviewProps {
+  // eslint-disable-next-line react/no-unused-prop-types
```
Comment (petorsong) [inline src/components/CCPDashboard/PatientOverview.tsx:30]: this prop is not being used - added a disable to investigate

File: src/containers/EventCreationPage.tsx [hunk 4]
```diff
@@ -45,21 +45,25 @@ const EventCreationPage = () => {
   const handleOpenDateModal = () => setOpenDateModal(true);
   const handleCloseDateModal = () => setOpenDateModal(false);
 
-  const handleNameChange = (e: any) => {
-    setEventName(e.target.value);
+  const handleNameChange = (e: React.ChangeEvent<HTMLElement>) => {
+    const target = e.target as HTMLInputElement;
+    setEventName(target.value);
   };
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (petorsong) [inline src/containers/EventCreationPage.tsx:52]: can't avoid this here cause implicit type casting happens

File: src/containers/PatientProfilePage.tsx [hunk 5]
```diff
@@ -43,26 +43,26 @@ const PatientProfilePage = ({
...
-    gender: "Male",
+    gender: 'Male',
     age: null,
-    notes: "",
+    notes: '',
     status: Status.ON_SITE,
     runNumber: null,
   });
 
   useEffect(() => {
```
Comment (petorsong) [inline src/containers/PatientProfilePage.tsx:64]: we get an error for this line: `React Hook useEffect has a missing dependency`
not sure if it's important

File: src/overrides-mui.d.ts [hunk 6]
```diff
@@ -1,4 +1,5 @@
-import { MuiPickersOverrides } from "@material-ui/pickers/typings/overrides";
+/* eslint-disable */
```
Comment (petorsong) [inline src/overrides-mui.d.ts:1]: this file has a lot of errors - may not be worth going through and trying to resolve all of them

File: src/containers/PatientProfilePage.tsx [hunk 7]
```diff
@@ -94,17 +94,17 @@ const PatientProfilePage = ({
   }, [data]);
```
Comment (wongcmichelle) [inline src/containers/PatientProfilePage.tsx:94]:
```suggestion
  }, [data, formFields, loading, mode]);
```

File: src/containers/PatientProfilePage.tsx (hunk 5 above)
Comment (wongcmichelle) [reply src/containers/PatientProfilePage.tsx:64]: this should be fixed

**Review by wongcmichelle — CHANGES_REQUESTED** [review]
Otherwise, lgtm :)

File: src/containers/PatientProfilePage.tsx (hunk 7 above)
Comment (petorsong) [reply src/containers/PatientProfilePage.tsx:94]: i tested this again on an edit patient page (http://localhost:3000/patients/edit/3/1) and i get into an infinite loop with the page going unresponsive
image

Comment (petorsong) [reply src/containers/PatientProfilePage.tsx:94]: moved this into a separate ticket - needs to be fixed before launch
https://www.notion.so/uwblueprintexecs/useEffect-error-7108d3eee3cf4c96893e3de358ae5a55

**Pull request merged**